src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -321,6 +325,56 @@ export const handleGetPublicForm: ControllerHandler<
           }
           return res.json({ form: publicForm, isIntranetUser })
         })
+    case FormAuthType.SGID_MyInfo: {
+      const accessTokenCookie = req.cookies[SGID_COOKIE_NAME]
+      if (!accessTokenCookie) {
+        return res.json({
+          form: publicForm,
+          isIntranetUser,
+        })
+      }
+      res.clearCookie(SGID_COOKIE_NAME)
+      res.clearCookie(MYINFO_LOGIN_COOKIE_NAME)
+      return SgidService.extractSgidJwtMyInfoPayload(accessTokenCookie)
+        .asyncAndThen((auth) =>
+          SgidService.retrieveUserInfo({ accessToken: auth.accessToken }),
+        )
+        .andThen((userInfo) => {
+          const data = new SGIDMyInfoData(userInfo.data)
+          return MyInfoService.prefillAndSaveMyInfoFields(
```

Comment:
If we reuse the same MyInfoService to prefill fields from sgid myinfo, are we guaranteed that the types of field supported in sgid myinfo is a subset of singpass myinfo (if no, will this be an issue)?

Reply:
Good catch. Right now this depends on the frontend UI choices that we make. We're thinking of using the same form fields for MyInfo for sgID-MyInfo, so sgid myinfo will indeed be a strict subset of singpass myinfo.

Reply:
Understand, thanks!
Off hand, I don't see any issue with one single service handling the prefill/saving of both MyInfo and sgid-Myinfo fields, but can we build it in a way that even if MyInfo and sgid-Myinfo are not subsets of another another, this still works (if not already the case)?

Then in future, if there are new fields introduced in sgid-Myinfo which are not in Myinfo, we only need to update the frontend UI to accomodate such fields